c_src/TLH_absTab.c

```diff
@@ -0,0 +1,225 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <glib.h>
+#include "absTab.h"
+#include "message.h"
+
+struct AbsTab {
+    /* a two-level hash table where the value of each 
+     * key points to another (level2) hash table 
+     */
+    GHashTable *lev1; /* directory watch descriptors */
+    GHashTableIter lev1_rmItr, lev2_rmItr; /* used for remove_next_absTab() */
+    int rmItr_initialized;
+    int *lev1_rmItr_key;
+    GHashTable *lev1_rmItr_val;
+    /* used for remove_next_absTab(). This is not very nice,
+     * but I need lev1_rmItr_key (directory wd) because glib
+     * does not provide any API for peeking keys.
+     */
+};
+
+AbsTab_t new_absTab()
+{
+    struct AbsTab *tlh = malloc(sizeof(struct AbsTab));
+    if(tlh == NULL)
+        return NULL;
+    tlh->lev1 = g_hash_table_new(g_int_hash, g_int_equal);
+    /* TODO: check for success */
+    tlh->lev1_rmItr_key = NULL;
+    tlh->lev1_rmItr_val = NULL;
+    tlh->rmItr_initialized = 0;
+    return tlh;
+}
+
+int destroy_absTab(AbsTab_t tab)
+{
+    GHashTableIter itr1;
+    int *itr1_key;
+    GHashTable *lev2_tab = NULL;
+    g_hash_table_iter_init(&itr1, tab->lev1);
+    /* destroy each level2 table */
+    while(g_hash_table_iter_next(&itr1, (void**) &itr1_key, (void**) &lev2_tab)) {
+        g_hash_table_destroy(lev2_tab);
+        /* TODO: Free memory associated to each element in lev2_tab. For now,
+         * we assume this is taken care of by the caller (user) of absTab.
+         */
+    }
+    /* destroy level1 table */
```

Comment:
This comment is pretty redundant with what the code does. In other words, it's better if your comment focuses on the why instead of the how. Generally speaking you want the code to be self-explanatory which usually involves having good variable and function names.